### PR TITLE
[NEXUS-8837] Add response status line to remote.storage.outbound

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8837

### CODE REVIEW

**Comment by jdillon** [comment]
pending updating NPM location

**Comment by mcculls** [comment]
+1 ... check with @peterlynch about the NPM changes as he was also looking at making these logging statements consistent (I believe HttpProxyMetadataTransport and HttpTarballTransport are the two places that do outbound logging)

**Comment by jdillon** [comment]
@mcculls yup, I'll update that once I get confirmation from @peterlynch this is what he wants.

**Comment by adreghiciu** [comment]
+1

**Comment by peterlynch** [comment]
Yes - lets update HttpProxyMetadataTransport and HttpTarballTransport

- remove NPM and NPMTarball words
- add timing information

NPM also logs before the request is sent, which is helpful to us in the case where we see a thread dump thread id and want to match that to an outgoing request currently active - HttpClientRemoteStorage could do the same.

File: components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientRemoteStorage.java [hunk 1]
```diff
@@ -511,16 +511,24 @@ HttpResponse executeRequest(final ProxyRepository repository, final ResourceStor
...
     try {
-      return doExecuteRequest(repository, request, httpRequest, contentRequest);
+      response = doExecuteRequest(repository, request, httpRequest, contentRequest);
     }
     finally {
       timerContext.stop();
       if (stopwatch != null) {
-        outboundRequestLog
-            .debug("[{}] {} {} - {}", repository.getId(), httpRequest.getMethod(), httpRequest.getURI(), stopwatch);
+        outboundRequestLog.debug("[{}] {} {} -> {}; {}",
```
Comment (peterlynch) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientRemoteStorage.java]: do we need the semi colon and arrow?

not against this new format, but just curious mostly why you added that and the arrow.

**Comment by peterlynch** [comment]
+1 if we make other places the logger is used the same format

File: components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientRemoteStorage.java [hunk 2]
```diff
@@ -511,16 +511,24 @@ HttpResponse executeRequest(final ProxyRepository repository, final ResourceStor
     if (outboundRequestLog.isDebugEnabled()) {
       stopwatch = new Stopwatch().start();
```
Comment (peterlynch) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientRemoteStorage.java]: can use non-deprecated Stopwatch.createStarted() instead while here

File: components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientRemoteStorage.java (hunk 1 above)
Comment (peterlynch) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientRemoteStorage.java]: fine with leaving this new format

**Comment by peterlynch** [comment]
pending if @jdillon wants to add a reusable outbound logging/metric component or not

File: components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientRemoteStorage.java (hunk 1 above)
Comment (jdillon) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientRemoteStorage.java]: separator between request -> response; and timeing

**Comment by jdillon** [comment]
+1 I'm not going to change this PR any further @peterlynch if this is what you want, merge it.

**Pull request merged**